File: unfuddle/ProjectEuler/clojure/Euler4.clj
(defn palindrome? [n] (= (seq (str n)) (reverse (str n))))

(defn solve-it []
  (reduce max
    (filter palindrome? (for [i (range 1000) :when (> i 99) j (range 1000) :when (>= j i)] (* i j)))
  )
)

(println (palindrome? 123))
(println (palindrome? 121))

(println (solve-it))
  

File: unfuddle/ProjectEuler/ruby/Euler4.rb
# Largest palindrome that is product of two 3-digit numbers.

def palindrome?(n)
  str = n.to_s
  return str == str.reverse
end

candidates = []
100.upto(999) do |n|
  n.upto(999) do |m|
    product = n * m
    candidates.push(product) if palindrome?(product)
  end
end

puts candidates.max